src/util/xhr/getBlobFromURL.js

```diff
@@ -10,34 +25,67 @@
  *    }
  * }>}
  */
-function getBlobFromURL(urlToLoad, max_timeout_ms) {
+function getBlobFromURL(urlToLoad, maxTimeoutMS) {
   return new Promise((resolve, reject) => {
-    let timerID = setTimeout(() => {
-      reject({
-        status: 'error',
-        message: 'Timeout loading Blob URL'
-      });
-    }, max_timeout_ms);
+    const timerID = rejectOnTimeout(maxTimeoutMS, reject);
 
-    let xhr = new XMLHttpRequest();
-    xhr.responseType = 'blob';
-    xhr.onload = function (response) {
-      clearTimeout(timerID); // clear timeout reject error
+    // If fetch exists, use it to fetch blob, otherwise use XHR.
+    // XHR causes issues on safari 14.1 so we prefer fetch
+    const fetchBlob = fetch ? loadUrlUsingFetch : loadUrlUsingXhr;
```

Comment:
I think window.fetch is safer, using just fetch might result in reference exception

See this example

<img width="374" alt="Screen Shot 2022-01-05 at 21 27 38" src="https://user-images.githubusercontent.com/59408474/148276999-65bd3b09-a6e1-4810-89fa-eb86ad6aceb0.png">
r

Reply:
@patrick-tolosa good point, I've updated this to be
(typeof fetch !== 'undefined' && fetch)